src/v0/destinations/ga4/customMappingsHandler.js

```diff
@@ -0,0 +1,156 @@
+const get = require('get-value');
+const {
+  validateEventName,
+  basicValidation,
+  isReservedEventName,
+  addClientDetails,
+  removeReservedParameterPrefixNames,
+  prepareUserConsents,
+  removeInvalidParams,
+  GA4_RESERVED_PARAMETER_EXCLUSION,
+  getGA4CustomParameters,
+  buildDeliverablePayload,
+  GA4_PARAMETERS_EXCLUSION,
+  prepareUserProperties,
+  mapWithJsonPath,
+} = require('./utils');
+const { InstrumentationError } = require('@rudderstack/integrations-lib');
+const {
+  removeUndefinedAndNullRecurse,
+  constructPayload,
+  isDefinedAndNotNull,
+  isEmptyObject,
+  removeUndefinedAndNullValues,
+  isHybridModeEnabled,
+  getIntegrationsObj,
+} = require('../../util');
+const { trackCommonConfig, ConfigCategory, mappingConfig } = require('./config');
+
+const findGA4Events = (eventsMapping, event) => {
+  // Find the event using destructuring and early return
+
+  const validMappings = eventsMapping.filter(
+    (mapping) =>
+      mapping.rsEventName?.trim().toLowerCase() === event.trim().toLowerCase() &&
+      mapping.ga4EventName,
+  );
+  // Return an empty object if event not found
+  return validMappings;
+};
+
+const handleCustomMappings = (message, Config) => {
+  const { eventsMapping } = Config;
+
```

Comment:
A minor change for better readability & removing redundant initialisation

let rsEvent = get(message, 'event');
  if (message.type.toString().toLowerCase() === 'track') {
    basicValidation(rsEvent);
  } else {
    // for events other than track we will search with $eventType
    // example $track / $page
    rsEvent = `$${get(message, 'type')}`;
  }

Reply:
Addressed